lib/supply/client.rb

```diff
@@ -18,38 +19,27 @@ class Client
     # @!group Login
     #####################################################
 
-    # Initializes the auth_client and api_client using the specified information
-    # @param path_to_key: The path to your p12 file
-    # @param issuer: Email addresss for oauth
-    # @param passphrase: Passphrase for the p12 file
-    def initialize(path_to_key: nil, issuer: nil, passphrase: nil)
-      passphrase ||= "notasecret"
+    # Initializes the android_publisher and its auth_client using the specified information
+    # If not specified, it tries to take the value of the secrets.json from ENV['GOOGLE_APPLICATION_CREDENTIALS']
+    # @param path_to_service_account_json: The path to your service account Json file
+    def initialize(path_to_service_account_json: nil)
+      # REVIEW: do we want to force override ? If not, we don't get the 'proper'
+      # printed table output
+      ENV['GOOGLE_APPLICATION_CREDENTIALS'] = File.expand_path path_to_service_account_json if path_to_service_account_json
 
-      key = Google::APIClient::KeyUtils.load_from_pkcs12(File.expand_path(path_to_key), passphrase)
+      scopes = [Androidpublisher::AUTH_ANDROIDPUBLISHER]
 
-      begin
-        self.auth_client = Signet::OAuth2::Client.new(
```

Comment:
It looks like it could be possible: https://github.com/google/google-auth-library-ruby/blob/master/lib/googleauth/service_account.rb#L57-L63

although this isn't the documented way, and the .p12 way it deprecated from Google's side anyway (and not recommended).